components/scroll-wrapper/scroll-wrapper.js

```diff
@@ -0,0 +1,222 @@
+			.d2l-scroll-wrapper-container {
+				box-sizing: border-box;
+				outline: none;
+				overflow-x: auto;
+			}
+			:host([h-scrollbar][show-actions]) .d2l-scroll-wrapper-container {
+				border-left: 1px dashed var(--d2l-color-mica);
+				border-right: 1px dashed var(--d2l-color-mica);
+			}
+			:host([dir="rtl"][scrollbar-right][show-actions]) .d2l-scroll-wrapper-container,
+			:host(:not([dir="rtl"])[scrollbar-left][show-actions]) .d2l-scroll-wrapper-container {
+				border-left: none;
+			}
+			:host([dir="rtl"][scrollbar-left][show-actions]) .d2l-scroll-wrapper-container,
+			:host(:not([dir="rtl"])[scrollbar-right][show-actions]) .d2l-scroll-wrapper-container {
+				border-right: none;
+			}
+
+			:host([dir="rtl"]) .d2l-scroll-wrapper-button-left,
+			.d2l-scroll-wrapper-button-right {
+				left: auto;
+				right: -10px;
+			}
+
+			:host([dir="rtl"]) .d2l-scroll-wrapper-button-right,
+			.d2l-scroll-wrapper-button-left {
+				left: -10px;
+				right: auto;
+			}
+
+			.d2l-scroll-wrapper-actions {
+				position: -webkit-sticky;
+				position: sticky;
+				top: 0;
+				z-index: 4;
+			}
+
+			.d2l-scroll-wrapper-button {
+				background-color: var(--d2l-color-regolith);
+				border: 1px solid var(--d2l-color-mica);
+				border-radius: 50%;
+				box-shadow: 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+				box-sizing: content-box;
+				cursor: pointer;
+				display: inline-block;
+				font-family: inherit;
+				font-size: inherit;
+				height: 18px;
+				line-height: 0;
+				margin: 0;
+				padding: 10px;
+				position: absolute;
+				text-decoration: none;
+				top: 4px;
+				width: 18px;
+			}
+			.d2l-scroll-wrapper-button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-scroll-wrapper-button:hover,
+			.d2l-scroll-wrapper-button:focus {
+				background-color: var(--d2l-color-sylvite);
+				outline-style: none;
+			}
+			.d2l-scroll-wrapper-button.focus-visible {
+				box-shadow: 0 0 0 2px #ffffff, 0 0 0 4px var(--d2l-color-celestine), 0 2px 12px 0 rgba(0, 0, 0, 0.15);
+			}
+			:host([scrollbar-right]) .d2l-scroll-wrapper-button-right {
+				display: none;
+			}
+			:host([scrollbar-left]) .d2l-scroll-wrapper-button-left {
+				display: none;
+			}
+
+			/* hide wrapper visuals from print view */
+			@media print {
+				.d2l-scroll-wrapper-actions {
+					display: none;
+				}
+				:host([h-scrollbar][show-actions]) .d2l-scroll-wrapper-container {
+					border-left: none;
+					border-right: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.showActions = false;
+		this._container = null;
+		this._hScrollbar = true;
+		this._scrollbarLeft = false;
+		this._scrollbarRight = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this._container = this.shadowRoot.querySelector('.d2l-scroll-wrapper-container');
+		new ResizeObserver(
+			() => requestAnimationFrame(
+				() => this.checkScrollbar()
+			)
+		).observe(this._container);
+		requestAnimationFrame(() => this.checkScrollbar());
```

Comment:
Is this needed? I did a quick check in Chrome and Safari and the observer seems to be called initially in both cases, but maybe I did not hit the scenario where it's needed.

Reply:
Yeah I'm seeing the same. I'll try it in Firefox & legacy-Edge and if it's always called initially I'll remove it. 👍

Reply:
Confirmed this wasn't needed -- removed.